providers/sh_cluster.rb

```diff
@@ -0,0 +1,44 @@
+# coding: UTF-8
+#
+# Cookbook Name:: cerner_splunk
+# Provider:: sh_cluster
+#
+
+use_inline_resources
+
+action :initialize do
+  search_heads = new_resource.search_heads
+  admin_password = new_resource.admin_password
+
+  execute 'Captain assignment' do
```

Comment:
That's because it was written before I knew about run state in chef, so getting the admin password is using a cat... and only root can read that file :)